corehq/apps/app_manager/translations.py

```diff
@@ -480,34 +497,43 @@ def update_case_list_translations(sheet, rows, app):
             ))
             continue
 
-        # The logic for updating a mapping and updating a MappingItem and a
-        # DetailColumn is almost the same. So, we smush the two together.
-        for index, translation_row in enumerate([row] + row.get("mappings", [])):
+        def _update_translation(row, language_dict, require_translation=True):
             ok_to_delete_translations = has_at_least_one_translation(
-                translation_row, 'default', app.langs)
+                row, 'default', app.langs
+            ) or not require_translation
```

Comment:
It would be more efficient to rephrase this so that the `has_at_least_one_translation` calculation (which iterates on row) is avoided if `require_translation` evaluates to false.
```python
ok_to_delete_translations = not require_translation or has_at_least_one_translation(...)
```